src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +70,20 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                for (Recipe recipe : recipes) {
+                    recipeKey = ((Keyed) recipe).getKey();
+                    if (!p.hasDiscoveredRecipe(recipeKey)) {
+                        unlocked = false;
```

Comment:
```suggestion
                        unlocked = false;
                        break;
```
use short circuiting evaluation (i think thats what its called)

Reply:
Okay, I will change it. Nice suggestion.